realm/src/androidTest/java/io/realm/RealmTest.java

```diff
@@ -1056,4 +1056,27 @@ public void testWriteEncryptedCopy() throws Exception {
         }
     }
 
+    public void testWrongKeyShouldFail() {
+        final String WRONG_KEY_REALM = "wrong-key-realm.realm";
+
+        // Wrong key size
+        try {
+            Realm.getInstance(getContext(), WRONG_KEY_REALM, new byte[63]);
+            fail();
+        } catch (IllegalArgumentException ignored) {
+        } catch (Exception ignored) {
```

Comment:
have we ever experienced this issue? It just complicates a lot of test cases if we have to do that in case someone change our tests. Surely if you change the tests, they work differently. Catching all exceptions in any testcase would make all our testcases fail? So then we need it for every single testcase - even the positive tests?